src/state.c

```diff
@@ -70,7 +70,8 @@ void backend_changed(struct xen_device *xendev, const char *node)
             }
         }
 
-        free(val);
+        if (val)
+            free(val);
```

Comment:
`free()` can take a NULL pointer, so this isn't really needed.

Reply:
agreed